core/src/main/scala/kafka/admin/ConfigCommand.scala

```diff
@@ -520,30 +545,48 @@ object ConfigCommand extends Config {
-      if (entityTypeVals.contains(ConfigType.Client) || entityTypeVals.contains(ConfigType.Topic) || entityTypeVals.contains(ConfigType.User))
+      if (entityTypeVals.contains(EntityType.Client) || entityTypeVals.contains(EntityType.Topic) || entityTypeVals.contains(EntityType.User))
         CommandLineUtils.checkRequiredArgs(parser, options, zkConnectOpt, entityType)
       if(options.has(alterOpt)) {
-        if (entityTypeVals.contains(ConfigType.User) || entityTypeVals.contains(ConfigType.Client) || entityTypeVals.contains(ConfigType.Broker)) {
+        if (entityTypeVals.contains(EntityType.User) || entityTypeVals.contains(EntityType.Client) || entityTypeVals.contains(EntityType.Broker)) {
           if (!options.has(entityName) && !options.has(entityDefault))
             throw new IllegalArgumentException("--entity-name or --entity-default must be specified with --alter of users, clients or brokers")
         } else if (!options.has(entityName))
-            throw new IllegalArgumentException(s"--entity-name must be specified with --alter of ${entityTypeVals}")
+            throw new IllegalArgumentException(s"--entity-name must be specified with --alter of $entityTypeVals")
 
         val isAddConfigPresent: Boolean = options.has(addConfig)
         val isDeleteConfigPresent: Boolean = options.has(deleteConfig)
         if(! isAddConfigPresent && ! isDeleteConfigPresent)
           throw new IllegalArgumentException("At least one of --add-config or --delete-config must be specified with --alter")
       }
+
+      val (allowedEntityTypes, connectOptString) = if (options.has(bootstrapServerOpt))
+        (EntityType.brokerSupportedTypes, "--bootstrap-server")
+      else
+        (EntityType.zkSupportedTypes, "--zookeeper")
+
       entityTypeVals.foreach(entityTypeVal =>
-        if (!ConfigType.all.contains(entityTypeVal))
-          throw new IllegalArgumentException(s"Invalid entity-type ${entityTypeVal}, --entity-type must be one of ${ConfigType.all}")
+        if (!allowedEntityTypes.contains(entityTypeVal))
+          throw new IllegalArgumentException(s"Invalid entity-type $entityTypeVal, --entity-type must be one of $allowedEntityTypes with the $connectOptString argument")
       )
       if (entityTypeVals.isEmpty)
         throw new IllegalArgumentException("At least one --entity-type must be specified")
-      else if (entityTypeVals.size > 1 && !entityTypeVals.toSet.equals(Set(ConfigType.User, ConfigType.Client)))
-        throw new IllegalArgumentException(s"Only '${ConfigType.User}' and '${ConfigType.Client}' entity types may be specified together")
+      else if (entityTypeVals.size > 1 && !entityTypeVals.toSet.equals(Set(EntityType.User, EntityType.Client)))
+        throw new IllegalArgumentException(s"Only '${EntityType.User}' and '${EntityType.Client}' entity types may be specified together")
     }
   }
+
+  object EntityType {
```

Comment:
My thinking was the following:
`ConfigType` is defined in `kafka.server` and is related to the back-end functionality - it is the names of the zNodes. The javadoc says this as well:
```
/**
 * Represents all the entities that can be configured via ZK
 */
```
The new config type I'm adding called `"broker-loggers"` is only used in the front-end - that tool is the only class that uses it. I thought it would be more confusing if we had it in kafka.server rather than just define it in the command itself.
Now, when it comes to defining it in the command we could just have a separate variable called `brokerLoggerConfigType` but I also wanted to add two lists for validating zk-only allowed configs and broker-only allowed configs:
```scala
    val zkSupportedTypes = Seq(Topic, Client, User, Broker)
    val brokerSupportedTypes = Seq(Broker, BrokerLogger)
```
Those make sense to be in the back-end though.

I also thought that coupling the front-end flags with the back-end is more brittle, even if we are super unlikely to ever change a zNode path.
I don't think it'll be too confusing for new people as they can see that the one is defined and used entirely in the tools class whereas the other `ConfigType` is used throughout.

Regardless of the wall of text, I don't feel strongly about this :D. Let me know what you think after I explained my reasoning

Reply:
In terms of reasoning:
- Both entities are in the backend, so I don't think there is coupling of front end and back end. ConfigCommand and DynamicConfigManager are both inside core Kafka module. If ConfigCommand was in client module, I'd ask you to create ConfigType in `common` and make sure they both use it. It is literally shared information - and all types of dynamic configs will remain long after ZK is gone.
- From what I see, this isn't a public API, so the Javadoc can change :) There's no real contract around this as far as I can tell.
- I liked the two lists for ZK and not ZK. It cleaned the validation nicely. They can even be in ConfigCommand, they just don't require EntityType...
- I don't think it is confusing. I just think it is redundant. And you had to make larger changes to add the new type, which always add a bit of risk.